### PR TITLE
[WIP] Unit test a React component

### PR DESCRIPTION
**Pull request by Kmaschta**
preview

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -109,6 +109,7 @@
     "redux-mock-store": "0.0.3",
     "selenium-standalone": "~4.9.0",
     "sinon": "2.0.0-pre",
+    "teaspoon": "~6.1.2",
```
Comment (fzaninotto) [inline package.json]: I know we've used `teaspoon` in the past, but isn't `enzyme` (https://github.com/airbnb/enzyme) the new standard? Ping @jeromemacias @brikou @RobinBressan @djhi

Comment (djhi) [inline package.json]: Haven't use any of them yet. I'll let the others decide

File: src/frontend/js/product/ProductItem.spec.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import chaiEnzyme from 'chai-enzyme';
+import React from 'react';
+
+import ProductItem from './ProductItem';
+
+
+describe('Component ProductItem', () => {
+    let props;
+
+    beforeEach(() => {
```
Comment (djhi) [inline src/frontend/js/product/ProductItem.spec.js]: Not needed, just initialize the props as a const at L10 and remove the `beforeEach` hook

File: src/frontend/js/product/ProductItem.spec.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        const wrapper = shallow(<ProductItem {...props} />);
+        assert.deepEqual(wrapper.find('p.card-text').text(), 'Best product ever');
+    });
+
+    it('should display the correct thumbnail', () => {
+        const wrapper = shallow(<ProductItem {...props} />);
+        expect(wrapper.find('img')).to.have.attr('src', 'http://google.fr/image.jpg');
+    });
+});
+
```
Comment (fzaninotto) [inline src/frontend/js/product/ProductItem.spec.js]: please add a test where you also trigger a click. You should pass a mock click handler and make sure it was clicked

File: src/frontend/js/product/ProductItem.spec.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+chai.use(chaiEnzyme());
```
Comment (fzaninotto) [inline src/frontend/js/product/ProductItem.spec.js]: please move that up, it's strange to have it at the bottom of the test

File: src/frontend/js/product/ProductItem.spec.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    });
+
+    it('should display the correct description', () => {
+        const wrapper = shallow(<ProductItem {...props} />);
+        assert.deepEqual(wrapper.find('p.card-text').text(), 'Best product ever');
+    });
+
+    it('should display the correct thumbnail', () => {
+        const wrapper = shallow(<ProductItem {...props} />);
+        expect(wrapper.find('img')).to.have.attr('src', 'http://google.fr/image.jpg');
```
Comment (fzaninotto) [inline src/frontend/js/product/ProductItem.spec.js]: don't mix `expect` and `assert`. Choose one, and stick to it.

File: package.json [hunk 6]
```diff
@@ -86,8 +86,10 @@
     "babel-preset-react": "~6.3.13",
     "babel-preset-stage-0": "~6.3.13",
     "chai": "~3.3.0",
+    "chai-enzyme": "~0.4.1",
+    "cheerio": "~0.20.0",
```
Comment (fzaninotto) [inline package.json]: not used

File: makefile [hunk 7]
```diff
@@ -111,7 +111,7 @@ test-api-functional: reset-test-database
 	@NODE_ENV=test NODE_PORT=3010 ./node_modules/.bin/mocha --require "./babel-transformer" --require=co-mocha --recursive ./e2e/api
 
 test-frontend-unit:
-	@NODE_ENV=test ./node_modules/.bin/mocha --compilers="css:./webpack/null-compiler,js:babel-core/register" --recursive ./src/frontend/js/**/*.spec.js
+	@NODE_ENV=test ./node_modules/.bin/mocha --require ./test/setupJsdom.js --compilers="css:./webpack/null-compiler,js:babel-core/register" --recursive ./src/frontend/js/**/*.spec.js
```
Comment (fzaninotto) [inline makefile]: I wonder if you can't enable it on a per-test basis, instead of for the entire suite.

File: package.json (hunk 6 above)
Comment (Kmaschta) [inline package.json]: It's a peer dep of [chai-enzyme](https://github.com/producthunt/chai-enzyme/blob/master/package.json)

**Comment by Kmaschta** [comment]
Actually, `chai-enzyme` isn't needed for this test, but it can be really useful for another tests.

**Comment by fzaninotto** [comment]
Then write another test where it's useful

**Pull request closed without merging**